SupportClient/SupportClient/Controllers/ViewController.swift

```diff
@@ -101,73 +183,98 @@ class ViewController: UIViewController {
-        return 75
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView == headerCollectionView {
+            return headersData.count
+        } else {
+            return searchController.isActive ? filteredFeedbackData.count : feedbackData.count
+        }
     }
     
-    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let _ = feedbackTableView.cellForRow(at: indexPath) as! FeedbackTableViewCell
-        var _ = feedbackData[indexPath.row]
-        // TODO: navigate to the messaging page
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == headerCollectionView {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: HeaderCollectionViewCell.reuseID, for: indexPath) as! HeaderCollectionViewCell
+            let header = headersData[indexPath.item]
+            cell.configure(with: header)
+            return cell
+        } else {
+            return UICollectionViewCell()
+        }
     }
+        
+}
+
+// MARK: - UICollectionView Delegate
+extension ViewController: UICollectionViewDelegate {
     
-    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
-        if editingStyle == .delete {
-            feedbackData.remove(at: indexPath.row)
-            feedbackTableView.deleteRows(at: [indexPath], with: .fade)
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if collectionView == feedbackCollectionView { }
+    }
+    
+}
+
+// MARK: - UICollectionView DelegateFlowLayout
+extension ViewController: UICollectionViewDelegateFlowLayout {
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        if collectionView == headerCollectionView {
+            return CGSize(width: (view.frame.width - 60) / 2, height: 40)
+        } else {
+            return CGSize()
         }
-        tableView.reloadData()
     }
     
 }
 
-// MARK: - TableViewDataSource
-extension ViewController: UITableViewDataSource {
+// MARK: - UISearchController Delegate
+extension ViewController: UISearchControllerDelegate {
+    
+    func willDismissSearchController(_ searchController: UISearchController) {
+        navigationController?.setNavigationBarHidden(false, animated: true)
+        setupConstraints()
+    }
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return feedbackData.count
+    func willPresentSearchController(_ searchController: UISearchController) {
+        navigationController?.setNavigationBarHidden(true, animated: true)
+        setupSearchingConstraints()
```

Comment:
sorry a picky thing -- i think there's a setting (`dimsBackground` something) that we can set to false so that the collection view doesn't gray out when the search bar is selected, to keep in line with the designs
& change the blue color of the "Cancel" button to match the gray in the designs

Reply:
`dimsBackground` is deprecated and became `obscuresBackground` in iOS 12.0, but will change